cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,376 @@
+    ]
+}
+  ```
+
+the user-specific details would be {ACCOUNT_ID} and {OIDC_PROVIDER}.
+
+
+#### Radius support
+
+**installation**
+
+User can enable IRSA for AWS by by choosing "IRSA" option by choosing IRSA during rad init and providing a RoleARN (`arn:aws:iam::123456789012:role/radius-role`). This stores the roleARN as credential. It also mounts the service-account token to ucp and rp pods as secret.
+
+
+```
+    Containers:
+      ucp:
+        Container ID:   docker://faa5af929f91e053b04b023e05c4fa6363e87688df265443d9f377258fc3fd04
+        Image:          ghcr.io/radius-project/ucpd:latest
+        Image ID:       docker-pullable://ghcr.io/radius-project/ucpd@sha256:991ed703a3260a19535690242cef4d7e4d076038b016443d9106f2d76d57b0da
+        Ports:          9443/TCP, 9090/TCP
+        Host Ports:     0/TCP, 0/TCP
+        :
+        Environment:
+          UCP_CONFIG:                   /etc/config/ucp-config.yaml
+          BASE_PATH:                    /apis/api.ucp.dev/v1alpha3
+          TLS_CERT_DIR:                 /var/tls/cert
+          PORT:                         9443
+        Mounts:
+          /etc/config from config-volume (rw)
+          /var/run/secrets/eks.amazonaws.com/serviceaccount from aws-iam-token (ro)
+          /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-d22ld (ro)
+          /var/tls/cert from cert (ro)
+```
+
+Note: Since AWS IRSA works by mounting the service-account token to pods, the decision to use IRSA should be taken at install time. Credentials can be registered later on.
+
+**post installation**
+
+Once the installation is complete, the user can use rad env update to store relevant information for deploying resources to accounts and region associated with specific radius environment.
+
+```
+rad env update qa --aws-account-id <aws-account-id> --aws-region <aws-region> 
+```
+
+At this point, we can use aws AssumeRole to manage resources using environment specific account information. rp (for recipes) and ucp will have code changes that make sure of aws assumerole to be able to use the right credential specific to the radius environment for managing the aws resources.  
+
+```
+  client := sts.NewFromConfig(cfg)
+  assumeRoleArn := retrieveRoleARNCredential() //has logic to get the roleARN stored as credential
+  // Assume the new role
+  assumeRoleProvider := stscreds.NewAssumeRoleProvider(client, assumeRoleArn)
+  // Create a new config with the new credentials
+  cfg.Credentials = assumeRoleProvider
+  cfg.Region = region
+  // Create a new S3 service client
+  s3client := s3.NewFromConfig(cfg)
+  // Call the ListBuckets function
+  output, err := s3client.ListBuckets(context.TODO(), &s3.ListBucketsInput{})
+  :
+
+```
+
+### API design
+
+```
+@doc("AWS credential kind")
+enum AWSCredentialKind {
+  @doc("The AWS Access Key credential")
+  AccessKey,
+
+  @doc("The AWS IRSA credential")
+  IRSA,
+}
+```
+
+```
+model AwsIRSACredentialProperties extends AwsCredentialProperties {
```

Comment:
We should be consistent about captialization of initialisms. I see `AWS` (go style) on line 261 and `Aws` (javascript style) on line 272.

Reply:
What's the plan for this?